File: Templates/DisjointSetUnionBySize.py
class DisjointSet:
    def __init__(self,n):
        self.n = n
        self.parent = [i for i in range(n+1)]
        self.size = [1]*(n+1)
    
    def getParent(self, node) -> int:
        if self.parent[node] == node:
            return node
        
        self.parent[node] = self.getParent(self.parent[node])
        return self.parent[node]


    def unionBySize(self, u, v):
        u = self.getParent(u)
        v = self.getParent(v)

        if u==v:
            return
        
        if self.size[u] < self.size[v]:
            self.parent[u] = v
            self.size[v]+=self.size[u]
        else:
            self.parent[v] = u
            self.size[u]+=self.size[v]

File: Templates/DisjointSetUnionBySize.cpp
#include <bits/stdc++.h>
using namespace std;

class DisjointSet {
private:
    vector<int> parent, size;
    int n;
public:
    DisjointSet(int n){
        this->n = n;
        for (int i=0;i<=n; ++i){
            parent.push_back(i);
            size.push_back(1);
        }
    }

    int findParent(int x){
        if (parent[x] != x){
            parent[x] = findParent(parent[x]);
        }
        return parent[x];
    }

    void unionBySize(int x, int y){
        x = findParent(x);
        y = findParent(y);
        if (x == y) return;

        if (size[x] < size[y]){
            parent[x] = y;
            size[y] += size[x];
        } else {
            parent[y] = x;
            size[x] += size[y];
        }
    }
};